driver/src/main/java/org/neo4j/driver/internal/DriverFactory.java

```diff
@@ -148,21 +133,23 @@ protected ChannelConnector createConnector( ConnectionSettings settings, Securit
     }
 
     private InternalDriver createDriver( URI uri, SecurityPlan securityPlan, BoltServerAddress address, ConnectionPool connectionPool,
-            EventExecutorGroup eventExecutorGroup, RoutingSettings routingSettings, RetryLogic retryLogic, MetricsProvider metricsProvider, Config config )
+                                         EventExecutorGroup eventExecutorGroup, RoutingSettings routingSettings, RetryLogic retryLogic,
+                                         MetricsProvider metricsProvider, Config config )
     {
         try
         {
             String scheme = uri.getScheme().toLowerCase();
-            switch ( scheme )
+
+            if ( isRoutingScheme( scheme ) )
+            {
+                return createRoutingDriver( securityPlan, address, connectionPool, eventExecutorGroup, routingSettings, retryLogic, metricsProvider, config );
+            }
+            else if ( !isRoutingScheme( scheme ) )
             {
-            case BOLT_URI_SCHEME:
                 assertNoRoutingContext( uri, routingSettings );
                 return createDirectDriver( securityPlan, address, connectionPool, retryLogic, metricsProvider, config );
-            case BOLT_ROUTING_URI_SCHEME:
-                return createRoutingDriver( securityPlan, address, connectionPool, eventExecutorGroup, routingSettings, retryLogic, metricsProvider, config );
-            default:
-                throw new ClientException( format( "Unsupported URI scheme: %s", scheme ) );
             }
+            throw new ClientException( format( "Unsupported URI scheme: %s", scheme ) );
```

Comment:
haha. I am not fooled by these lines :D 
If-else should be good enough here as we've validated the scheme `Scheme.validateScheme( uri.getScheme() );`.

```
if( isRoutingScheme(scheme) ){...}
else {...}
```